statsbombpy/helpers.py

```diff
@@ -49,6 +49,38 @@ def reduce_events(all_events: dict, fmt: str) -> dict:
     return reduced_events
 
 
+def merge_events_and_frames(
+    events: dict,
+    frames: list,
+    drop_keys=[
+        "event_uuid",
+        "visible_area",
+        "match_id",
+        "freeze_frame",
+        "visible_player_counts",
+        "distances_from_edge_of_visible_area",
+    ],
+) -> dict:
+    # Convert frames list to dict
+    frames = {frame["event_uuid"]: frame for frame in frames}
+
+    # Merge frame metrics into events
+    events = {k: {**v, **frames.get(k, {})} for k, v in events.items()}
+
+    # Clean up visible player counts and cull unnecessary fields
+    for event_id, event in events.items():
+        for key in list(event):
```

Comment:
The loop is extracting some information from player counts and adding it into the events dict before deleting it, so we're not throwing away any tidying.
I tried refactoring it to do the extraction more cleanly and pruning with a comprehension at the end, but the code ended up less readable, especially since the loop is needed anyway.
I ended up leaving the function mostly intact (with a tiny bit of cleanup)

```    for _, event in events.items():
        visible_player_counts = events.get("visible_player_counts")
        if visible_player_counts:
            for team in visible_player_counts:
                if team["team_id"] == event["team"]["id"]:
                    event["visible_teammates"] = team["count"]
                else:
                    event["visible_opponents"] = team["count"]
    events = {
        ev_id: {k: event[k] for k in event.keys() if k not in drop_keys}
        for ev_id, event in events.items()
    }```

Reply:
yeah I can see that. I see that we're adding visible teamates and visible opponnents. Thats me not being able to read.

fair enough. that fine for me then.